projects/components/src/navigation/navigation-list.component.ts

```diff
@@ -94,36 +101,49 @@ export class NavigationListComponent implements OnChanges {
     private readonly featureStateResolver: FeatureStateResolver
   ) {}
 
-  public ngOnChanges(): void {
-    this.navItems = this.navListComponentService.resolveFeaturesAndUpdateVisibilityForNavItems(this.navItems);
-
-    this.navItems$ = this.featureStateResolver
-      .getCombinedFeatureState([ApplicationFeature.PageTimeRange, ApplicationFeature.NavigationRedesign])
-      .pipe(
-        map(featureState => featureState === FeatureState.Enabled),
-        switchMap(usePageLevelTimeRange => {
-          if (usePageLevelTimeRange) {
-            return this.navListComponentService.resolveNavItemConfigTimeRanges(this.navItems);
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.navItems) {
+      this.navItems = this.navListComponentService.resolveFeaturesAndUpdateVisibilityForNavItems(this.navItems);
+
+      // If the FFs are enabled, decorate the nav items with the corresponding time ranges.
+      // The time ranges in nav items are streams that get the most recent value from page time range preference service
+      this.navItems$ = this.featureStateResolver
+        .getCombinedFeatureState([ApplicationFeature.PageTimeRange, ApplicationFeature.NavigationRedesign])
+        .pipe(
+          map(featureState => featureState === FeatureState.Enabled),
+          switchMap(usePageLevelTimeRange => {
+            if (usePageLevelTimeRange) {
+              return this.navListComponentService.resolveNavItemConfigTimeRanges(this.navItems);
+            }
+
+            return of(this.navItems);
+          }),
+          shareReplay()
+        );
+
+      // For each nav item, find the (possibly new) active nav item
+      // Time range is set on component load, when it isn't already
+      // If page level TR is available because FF is enabled, uses that, otherwise the default
+      this.activeItem$ = combineLatest([
+        this.navItems$,
+        this.navigationService.navigation$.pipe(startWith(this.navigationService.getCurrentActivatedRoute()))
+      ]).pipe(
+        map(([navItems], index) => {
+          const activeItem = this.findActiveItem(navItems);
+          if (index === 0) {
+            const timeRangeQueryParamValue = this.navigationService.getQueryParameter(
+              NavigationListComponent.TIME_RANGE_QUERY_PARAM_KEY,
```

Comment:
We don't need to leak this impl detail or use index I don't believe - can we instead check with TR service if time range is assigned, and if not, set default? That way the query param logic remains only in that service and this logic works across multiple left nav loads/changes (every time ngOnChanges fires, we'll get a new index 0)